src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp

```diff
@@ -187,9 +243,63 @@ namespace AppInstaller::Manifest
         // Common fields across versions
         std::vector<FieldProcessInfo> result =
         {
-            { "InstallerType", [this](const YAML::Node& value)->ValidationErrors { m_p_installer->BaseInstallerType = ConvertToInstallerTypeEnum(value.as<std::string>()); return {}; } },
-            { "PackageFamilyName", [this](const YAML::Node& value)->ValidationErrors { m_p_installer->PackageFamilyName = value.as<std::string>(); return {}; } },
-            { "ProductCode", [this](const YAML::Node& value)->ValidationErrors { m_p_installer->ProductCode = value.as<std::string>(); return {}; } },
+            {
+                "InstallerType",
+                [forRootFields](const YAML::Node& value, std::any& any)->ValidationErrors

```

Comment:
This seems fragile, I would really prefer to avoid a capture.  But I'm guessing it isn't nearly as easy to do as it is to say.

Reply:
I tried splitting `GetRootFieldProcessInfo` and calling ValidateAndProcessFields with root, root installer and root localization separately. but that will result in several `ManifestError::FieldUnknown` warning. I don't want to deal with looking at the errors and remove them.

I removed the capture, but it still needs to any_cast to manifest or installer/localization. I'm not sure if the capture is better than this try catch approach.